src/components/search-history/search-history.component.tsx

```diff
@@ -1,55 +1,225 @@
+  DOWNLOAD,
+  DELETE,
+}
+
+export const SearchHistory: React.FC<SearchHistoryProps> = ({
+  isHistoryUpdated,
+  setIsHistoryUpdated,
+}) => {
   const { t } = useTranslation();
+  const [searchResults, setSearchResults] = useState<SearchHistoryItem[]>([]);
+  const [selectedSearchItemId, setSelectedSearchItemId] = useState(0);
+  const [cohortName, setCohortName] = useState("");
+  const [cohortDescription, setCohortDescription] = useState("");
+  const [queryName, setQueryName] = useState("");
+  const [queryDescription, setQueryDescription] = useState("");
+  const [page, setPage] = useState(1);
+  const [pageSize, setPageSize] = useState(10);
+  const [isClearHistoryModalVisible, setIsClearHistoryModalVisible] =
+    useState(false);
+  const [isDeleteCohortModalVisible, setIsDeleteCohortModalVisible] =
+    useState(false);
+  const [isSaveCohortModalVisible, setIsSaveCohortModalVisible] =
+    useState(false);
+  const [isSaveQueryModalVisible, setIsSaveQueryModalVisible] = useState(false);
+
+  useEffect(() => {
+    if (isHistoryUpdated) {
+      setSearchResults(getSearchHistory());
+      setIsHistoryUpdated(false);
+    }
+  }, [isHistoryUpdated, setIsHistoryUpdated]);
+
+  const handlePagination = ({ page, pageSize }: PaginationData) => {
+    setPage(page);
+    setPageSize(pageSize);
+  };
 
   const headers = [
     {
       key: "id",
       header: "#",
     },
     {
-      key: "query",
+      key: "description",
       header: t("query", "Query"),
     },
     {
       key: "results",
       header: t("results", "Results"),
     },
-    {
-      key: "query-definition",
-      header: t("query-definition", "Query Definition Options"),
-    },
-    {
-      key: "cohort-definition",
-      header: t("cohort-definition", "Cohort Definition Options"),
-    },
   ];
 
+  const clearHistory = () => {
+    window.sessionStorage.removeItem("openmrsHistory");
+    setSearchResults([]);
+    setIsClearHistoryModalVisible(false);
+  };
+
+  const saveCohort = async () => {
+    const cohortMembers: number[] = [];
+    const { patients } = searchResults[selectedSearchItemId];
+    patients.forEach((patient: Patient) =>
+      cohortMembers.push(parseInt(patient.id))
+    );
+    let cohort: Cohort = {
+      display: cohortName,
+      memberIds: cohortMembers,
+      description: cohortDescription,
+      name: cohortName,
+    };
+
+    try {
+      await createCohort(cohort);
+      setCohortName("");
+      setCohortDescription("");
+      setIsSaveCohortModalVisible(false);
+      showNotification({
+        title: t("cohortCreateSuccess", "Success"),
+        kind: "success",
+        critical: true,
+        description: "the cohort is saved",
+      });
+    } catch (error) {
+      showNotification({
+        title: t("cohortCreateError", "Error creating the cohort"),
+        kind: "error",
+        critical: true,
+        description: error?.message,
+      });
+    }
+  };
+
+  const handleDeleteSearchItem = async () => {
```

Comment:
It's better to take these logics out from the component to make it simple and testable.